torch/_inductor/config.py

```diff
@@ -259,7 +259,7 @@ def autotune_remote_cache_default() -> Optional[bool]:
 ]
 
 # enable operator reordering for peak memory optimization
-reorder_for_peak_memory = os.environ.get("TORCHINDUCTOR_REORDER_FOR_PEAK_MEMORY") == "1"
```

Comment:
it's easiest to do via ghstack but i think you can also push to origin/main or something I forget exactly. would recommend ghstack.

If you've already submitted this pr, the easiest to do that is just to `git reset --soft HEAD~1; git commit -m "tmp" ; ghstack` and benchmark this commit as another pr.

Reply:
Let me try to figure this out.

Reply:
Sorry one other thing - we should test out the various implementations on ci runs as well, if possible. unless you already have clear idea that one is pareto optimal (compilation time, average memory change, worst memory change).

Reply:
I submitted a run [here](https://github.com/pytorch/pytorch/actions/runs/11236966162). Seems that it takes a few days for the results to show up in the [dashboard](https://hud.pytorch.org/benchmark/compilers). I will report back once the results show up.

Reply:
> if we are at all worried about riskiness of change it can also make sense to add a JK for enablement, or enable in OSS first then fbcode

I was initially very concerned about this, but as I am trying this flag on for more internal models, I am becoming less concerned. Though there are many potential situations I am not aware of, so I think it would be the best to leave this decision to the PyTorch team. You guys are are the experts here :)

Reply:
Seems all the dashboard run are done. This is the screenshot for the the summary (seems nothing is signficant). 

<img width="1706" alt="image" src="https://github.com/user-attachments/assets/94746f74-c84d-4a30-8382-56a1d95f698a">

Individual breakdowns can be found here:
* gh/xuanzhang816/1/head -- using all three heuristics  
  * [torchbench](https://hud.pytorch.org/benchmark/torchbench/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/1/head&lCommit=bba3dc40884cc19a322967c3e063b2ae6137ff06&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [huggingface](https://hud.pytorch.org/benchmark/huggingface/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/1/head&lCommit=bba3dc40884cc19a322967c3e063b2ae6137ff06&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [timm](https://hud.pytorch.org/benchmark/timm_models/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/1/head&lCommit=bba3dc40884cc19a322967c3e063b2ae6137ff06&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
* gh/xuanzhang816/2/head -- lpmf only
  * [torchbench](https://hud.pytorch.org/benchmark/torchbench/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/2/head&lCommit=a8aecf21613050520faec1dd3fedbc897d5367a4&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [huggingface](https://hud.pytorch.org/benchmark/huggingface/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/2/head&lCommit=a8aecf21613050520faec1dd3fedbc897d5367a4&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [timm](https://hud.pytorch.org/benchmark/timm_models/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/2/head&lCommit=a8aecf21613050520faec1dd3fedbc897d5367a4&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
* gh/xuanzhang816/3/head -- bfs only
  * [torchbench](https://hud.pytorch.org/benchmark/torchbench/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/3/head&lCommit=d5930e873a7333e0ad23a212e04259a60012efee&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [huggingface](https://hud.pytorch.org/benchmark/huggingface/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/3/head&lCommit=d5930e873a7333e0ad23a212e04259a60012efee&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [timm](https://hud.pytorch.org/benchmark/timm_models/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/3/head&lCommit=d5930e873a7333e0ad23a212e04259a60012efee&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
* gh/xuanzhang816/4/head -- dfs only
  * [torchbench](https://hud.pytorch.org/benchmark/torchbench/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/4/head&lCommit=ef2ffc51bebc4aedb85f889b033d8d0f1f9f8619&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [huggingface](https://hud.pytorch.org/benchmark/huggingface/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/4/head&lCommit=ef2ffc51bebc4aedb85f889b033d8d0f1f9f8619&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)
  * [timm](https://hud.pytorch.org/benchmark/timm_models/inductor_with_cudagraphs?dashboard=torchinductor&startTime=Fri,%2004%20Oct%202024%2018:44:25%20GMT&stopTime=Fri,%2011%20Oct%202024%2018:44:25%20GMT&granularity=hour&mode=training&dtype=amp&deviceName=cuda%20(a100)&lBranch=gh/xuanzhang816/4/head&lCommit=ef2ffc51bebc4aedb85f889b033d8d0f1f9f8619&rBranch=main&rCommit=59cdd8ddf15c76bb1da05cab6381c40efefaef6f)